### DISCUSSION TITLE
Replace play-json with nanojson in kamon-datadog?

### DISCUSSION THREAD

**Discussion by dvgica** [kamon-io/Kamon#888] [Ideas]
Hey all,

The kamon-datadog module [relies](https://github.com/kamon-io/Kamon/blob/master/build.sbt#L412) on play-json for serialization. It appears to be the only such module, and I'm wondering if we should try to consolidate on a single JSON library. The `kamon-akka-http` module uses [json4s](https://github.com/kamon-io/Kamon/blob/master/build.sbt#L412) so that might be a good candidate.

It does look like `nanojson` is [also used](https://github.com/kamon-io/Kamon/blob/master/build.sbt#L412). But, json4s at least makes it a bit easier to switch backends if needed.

This stems from a dependency conflict I'm having with the `play-json` pulled in by Kamon, and one already in my app.

Thoughts?

**Comment by dvgica** [comment]
Whoops, just realized that `json4s` is being used in a test context. So maybe `nanojson` would be better.

**Reply by ivantopo** [reply]
I don't see any particular reason not to switch the library. Our general stance is that the less libraries we have as dependencies, the better. The dependency on `play-json` was initially brought in by [this](https://github.com/kamon-io/kamon-datadog/pull/29) PR.

@stoiev do you maybe face any issues with using other json libraries or see any inconvenience in changing it?

**Reply by dvgica** [reply]
Cool. The replacement with `nanojson` looks pretty straight-forward to me, there's a very small amount of JSON to write. I can write a PR if there's no concerns about the idea.

**Reply by stoiev** [reply]
I see no issue/inconvenience. This lib was choose primarily because is under `play` umbrella - which I believed to be well known/maintained. If there is another json library already in use in Kamon, I agree that it's better to remove it and avoid extra bloat.

**Reply by dvgica** [reply]
PR up here, feedback welcome! https://github.com/kamon-io/Kamon/pull/892
